src/Sylius/Bundle/ResourceBundle/Doctrine/ORM/TreeRepository.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\ResourceBundle\Doctrine\ORM;
+
+use Doctrine\ORM\EntityManager;
+use Doctrine\ORM\Mapping\ClassMetadata;
+use Doctrine\ORM\QueryBuilder;
+use Gedmo\Tree\Traits\Repository\ORM\NestedTreeRepositoryTrait;
+
+/**
+ * @author Aram Alipoor <aram.alipoor@gmail.com>
+ */
+class TreeRepository extends EntityRepository
```

Comment:
This can be removed. I don't like Gedmo implementations and as I noticed right now we're only using `getChildren` and `getRootNodes` methods.

`getChildren` can be easily implemented (see [comment](https://github.com/Sylius/Sylius/pull/3773/files#r50017102)).
`getRootNodes` even easier (get all taxons without parent)

If we will need something more complex we can consider using it, but right now it seems like an overhead.